MeetupPlatformApi/Authentication/AuthenticationManager.cs

```diff
@@ -21,27 +21,54 @@ public CurrentUserInfo GetCurrentUserInfo(ClaimsPrincipal user)
         var idClaim = user.Claims.Single(claim => claim.Type == ClaimTypes.NameIdentifier);
         var id = Guid.Parse(idClaim.Value);
 
-        return new() {UserId = id};
+        return new() { UserId = id };
     }
 
-    public string IssueAccessToken(UserEntity user) =>
-        IssueToken(
+    public Guid GetNameIdentifier(string refreshToken)
+    {
+        var token = tokenHandler.ReadJwtToken(refreshToken);
+        return Guid.Parse(token.Claims.Single(claim => claim.Type == "nameid").Value);
+    }
+
+    public DateTime GetExpires(string refreshToken)
+    {
+        var token = tokenHandler.ReadJwtToken(refreshToken);
+        return token.ValidTo;
+    }
+
+    public TokenPair IssueTokenPair(UserEntity user)
+    {
+        var accessToken = IssueToken(
             payload: new Dictionary<string, object>
             {
                 {ClaimTypes.NameIdentifier, user.Id}
             },
-            lifetime: configuration.AccessTokenLifetime);
+            tokenLifetime: configuration.AccessTokenLifetime);
+
+        var refreshToken = IssueToken(
+            payload: new Dictionary<string, object>
+            {
+                {ClaimTypes.NameIdentifier, Guid.NewGuid() }
+            },
+            tokenLifetime: configuration.RefreshTokenLifetime);
+
+        return new TokenPair
+        {
+            AccessToken = accessToken,
+            RefreshToken = refreshToken
+        };
+    }
 
-    private string IssueToken(IDictionary<string, object> payload, TimeSpan lifetime)
+    private string IssueToken(IDictionary<string, object> payload, TimeSpan tokenLifetime)
```

Comment:
Consider renaming either  `payload` -> `tokenPayload` or `tokenLifetime` -> `lifetime`.

Reply:
I think it's strangely to add 'token' in payload variable name, but delete 'token' from lifetime variable name.
What about change 'tokenLifetime' ->'lifetime'?

Reply:
I was saying that we have 2 parameters, that both related to tokens, but only one of them has `token` prefix. We either need to either remove that prefix from the `tokenLifetime` parameter (so neither of parameters has a prefix) or add this prefix to the `payload` parameter (so both parameters have prefixes).